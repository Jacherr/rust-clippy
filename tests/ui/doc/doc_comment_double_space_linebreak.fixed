#![feature(custom_inner_attributes)]
#![rustfmt::skip]

#![warn(clippy::doc_comment_double_space_linebreak)]
#![allow(unused)]

//! Should warn on double space linebreaks \
//! in file/module doc comment

/// Should not warn on single-line doc comments
fn single_line() {}

/// Should not warn on single-line doc comments
/// split across multiple lines
fn single_line_split() {}

// Should not warn on normal comments

// note: cargo fmt can remove double spaces from normal and block comments
// Should not warn on normal comments  
// with double spaces at the end of a line

fn normal_comment() {
    /*
       Should not warn on block comments
    */

    /*
        Should not warn on block comments  
        with double space at the end of a line
     */
}

/// Should warn when doc comment uses double space \
/// as a line-break, even when there are multiple \
/// in a row
fn double_space_doc_comment() {}

/// Should not warn when back-slash is used \
/// as a line-break
fn back_slash_doc_comment() {}

fn main() {}
