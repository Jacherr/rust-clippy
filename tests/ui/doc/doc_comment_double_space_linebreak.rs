#![feature(custom_inner_attributes)]
#![rustfmt::skip]

#![warn(clippy::doc_comment_double_space_linebreak)]
#![allow(unused)]

//! Should warn on double space linebreaks  
//! in file/module doc comment

/// Should not warn on single-line doc comments
fn single_line() {}

/// Should not warn on single-line doc comments
/// split across multiple lines
fn single_line_split() {}

// Should not warn on normal comments

// note: cargo fmt can remove double spaces from normal and block comments
// Should not warn on normal comments  
// with double spaces at the end of a line  

#[doc = "This is a doc attribute, which should not be linted"]
fn normal_comment() {
    /*
       Should not warn on block comments
    */

    /*
        Should not warn on block comments  
        with double space at the end of a line
     */
}

/// Should warn when doc comment uses double space  
/// as a line-break, even when there are multiple  
/// in a row
fn double_space_doc_comment() {}

/// Should not warn when back-slash is used \
/// as a line-break
fn back_slash_doc_comment() {}

/// ðŸŒ¹ are ðŸŸ¥  
/// ðŸŒ· are ðŸŸ¦  
/// ðŸ“Ž is ðŸ˜Ž  
/// and so are ðŸ«µ  
/// (hopefully no formatting weirdness linting this)
fn multi_byte_chars_tada() {}

macro_rules! macro_that_makes_function {
   () => {
      /// Shouldn't lint on this!  
      /// (hopefully)
      fn my_macro_created_function() {}
   }
}

macro_that_makes_function!();

fn main() {}
