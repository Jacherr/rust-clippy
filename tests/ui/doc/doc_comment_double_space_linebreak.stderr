error: doc comments should use a back-slash (\) instead of a double space to indicate a linebreak
  --> tests/ui/doc/doc_comment_double_space_linebreak.rs:7:43
   |
LL |   //! Should warn on double space linebreaks  
   |  ___________________________________________^
LL | | //! in file/module doc comment
   | |____^
   |
   = note: `-D clippy::doc-comment-double-space-linebreak` implied by `-D warnings`
   = help: to override `-D warnings` add `#[allow(clippy::doc_comment_double_space_linebreak)]`
help: replace this double space with a back-slash
   |
LL ~ //! Should warn on double space linebreaks\
LL ~ //! in file/module doc comment
   |

error: doc comments should use a back-slash (\) instead of a double space to indicate a linebreak
  --> tests/ui/doc/doc_comment_double_space_linebreak.rs:35:51
   |
LL |   /// Should warn when doc comment uses double space  
   |  ___________________________________________________^
LL | | /// as a line-break, even when there are multiple  
LL | | /// in a row
   | |____^
   |
help: replace this double space with a back-slash
   |
LL ~ /// Should warn when doc comment uses double space\
LL ~ /// as a line-break, even when there are multiple\
LL ~ /// in a row
   |

error: doc comments should use a back-slash (\) instead of a double space to indicate a linebreak
  --> tests/ui/doc/doc_comment_double_space_linebreak.rs:44:12
   |
LL |   /// 🌹 are 🟥  
   |  ______________^
LL | | /// 🌷 are 🟦  
LL | | /// 📎 is 😎  
LL | | /// and so are 🫵  
LL | | /// (hopefully no formatting weirdness linting this)
   | |____^
   |
help: replace this double space with a back-slash
   |
LL ~ /// 🌹 are 🟥\
LL ~ /// 🌷 are 🟦\
LL ~ /// 📎 is 😎\
LL ~ /// and so are 🫵\
LL ~ /// (hopefully no formatting weirdness linting this)
   |

error: aborting due to 3 previous errors

